WordPress/Classes/ViewRelated/Post/Search/PostSearchViewModel.swift

```diff
@@ -86,27 +86,42 @@ final class PostSearchViewModel: NSObject, PostSearchServiceDelegate {
     private func reload(with notification: Foundation.Notification) {
         guard let userInfo = notification.userInfo else { return }
 
-        let existingObjects = Set(posts)
-        var updated = (userInfo[NSUpdatedObjectsKey] as? Set<NSManagedObject>) ?? []
-        var deleted = (userInfo[NSDeletedObjectsKey] as? Set<NSManagedObject>) ?? []
+        // The list displays the latest versions of a post when available,
+        // but it uses the original posts as identifiers (because they are stable).
+        // This method ensures that the list updates whenever either the
+        // original version or the latest version changes.
+        let existingOriginalPosts = Set(posts)
+        var existingLatestPosts: [NSManagedObject: NSManagedObject] = [:]
+        for object in posts {
+            existingLatestPosts[object] = object.latest()
+        }
 
-        updated.formIntersection(existingObjects)
-        deleted.formIntersection(existingObjects)
+        let updatedObjects = (userInfo[NSUpdatedObjectsKey] as? Set<NSManagedObject>) ?? []
```

Comment:
It's not ideal that it had to deviate from the Posts list implementation, but it's actually simpler than what the list does.